org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/generator/TestLanguageGenerator.xtend

```diff
@@ -24,6 +24,10 @@ import org.eclipse.xtext.ide.tests.testlanguage.testLanguage.Property
 class TestLanguageGenerator extends AbstractGenerator {
 
 	override void doGenerate(Resource resource, IFileSystemAccess2 fsa, IGeneratorContext context) {
+		if (fsa.getURI('').scheme == 'inmemory') {
```

Comment:
I think we should disable code generation in language server mode in general using a NEVER impl of `org.eclipse.xtext.generator.IShouldGenerate`.